### PR TITLE
Split shortcut menu into shortcuts and step types

### LINKED ISSUE #268: Step types should be separate from other shortcuts
image

### PR DESCRIPTION
**Pull request by theis999**
- Added new menu Step types
- Moved step types shortcuts to new menu.
- Added step types shortcuts for those that were missing. 
- Added new shortcut to select the search field.
- Added right click handlers to change step and delete step. 
- Added shortcut that match right click handlers of buttons. 
- Added new page in shortcut changer to handle the new menu.
- Added tooltips to buttons Step panels that didn't have them
- Fixed search columns, oversight with modifiers.

image
image
image
image

### CODE REVIEW

File: src/cMain.cpp [hunk 1]
```diff
@@ -788,7 +821,38 @@ void cMain::OnDeleteStepClicked(wxCommandEvent& event)
...
+	if (startRow < grid_steps->GetNumberRows())
+	{
+		grid_steps->SelectRow(startRow);
+	}
+	else if (startRow - 1 >= 0)
+	{
+		grid_steps->SelectRow(startRow - 1);
+	}
+
+	no_changes = false;
```
Comment (MortenTobiasNielsen) [inline src/cMain.cpp]: There is a lot of copy paste here from OnDeleteStepClicked - can you consolidate it a little?

File: src/cMain.cpp [hunk 2]
```diff
@@ -732,6 +733,35 @@ void cMain::OnChangeStepClicked(wxCommandEvent& event)
...
+	stack.Push({
+		.row = row,
+		.type = T_MODIFY,
+		.rows = ChangeStep(row, stepParameters)
+	});
+
+	grid_steps->SelectRow(row);
+	no_changes = false;
+}
+
```
Comment (MortenTobiasNielsen) [inline src/cMain.cpp]: There is a lot of copy paste here from OnChangeStepClicked - can you consolidate it a little?

**Review by MortenTobiasNielsen — CHANGES_REQUESTED** [review]
It seems like a good amount of the step types are still in the shortcuts tab:
image

File: src/cMain.cpp (hunk 1 above)
Comment (theis999) [reply src/cMain.cpp]: I can try at least.

Comment (theis999) [reply src/cMain.cpp]: Done

File: src/cMain.cpp (hunk 2 above)
Comment (theis999) [reply src/cMain.cpp]: Done

**Comment by theis999** [comment]
The issue you are seeing is essentially a migration issue. That should be fixed now.

File: src/cMain.cpp [hunk 3]
```diff
@@ -1663,7 +1703,19 @@ void cMain::OnMenuAutoCloseSaveAsClicked(wxCommandEvent& event)
...
+		OnChangeStepClicked(event);
+	else
+		wxMessageBox("Please select 1 row to change", "Selection not valid");
+	event.Skip();
+}
+
+void cMain::OnChangeAltMenuSelected(wxCommandEvent& event)
+{
+	if (btn_change_step->IsEnabled())
+		OnChangeStepClicked(event);
```
Comment (theis999) [inline src/cMain.cpp]: This appears to wrong

Comment (MortenTobiasNielsen) [reply src/cMain.cpp]: I what way?

**Comment by MortenTobiasNielsen** [comment]
I am not able to build the branch. I have tried to delete it locally and pull it again, but it didn't solve the problem.
image

File: src/cMain.cpp (hunk 3 above)
Comment (theis999) [reply src/cMain.cpp]: It should use the right click method. Otherwise the key bind won't work properly.

**Comment by theis999** [comment]
It is appears to have been a migration issue. Leaving in 2 function declarations. They should be fixed now, sorry. I won't use github's merge tool again.

**Review by MortenTobiasNielsen — APPROVED** [review]

**Pull request merged**